`cargo doc` fails if lending-iterator is depended on with `default-features = false`

I have the following dependency in my `Cargo.toml`: 

```toml
lending-iterator = { version = "0.1.6", default-features = false }
```

When I run `cargo doc` all the dependencies docs are built (not just my own crate). This result in the following error however:

```
error[E0433]: failed to resolve: could not find `alloc` in the list of imported crates
 --> /home/arvid/.cargo/registry/src/index.crates.io-6f17d22bba15001f/lending-iterator-0.1.6/src/lending_iterator/dyn/tests.rs:3:7
  |
3 |     ::alloc::{
  |       ^^^^^ could not find `alloc` in the list of imported crates

error: Compilation failed, aborting rustdoc
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -38,6 +38,8 @@ better-docs = [ # allowed to break MSRV
     "macro-vis",
 ]
 
+testing = []
+
 ui-tests = [
     "better-docs",
 ]
@@ -59,6 +61,8 @@ path = "src/proc_macros"
 version = "0.1.6"  # Keep in sync
 
 [dev-dependencies]
+lending-iterator.path = "."
+lending-iterator.features = ["testing"]
 
 [workspace]
 members = [

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -7,7 +7,7 @@ name = "lending-iterator"
 authors = [
     "Daniel Henry-Mantilla <daniel.henry.mantilla@gmail.com>",
 ]
-version = "0.1.7-rc1"  # Keep in sync
+version = "0.1.7"  # Keep in sync
 edition = "2021"
 
 license = "Zlib OR MIT OR Apache-2.0"
@@ -58,7 +58,7 @@ macro-vis.version = "0.1.1"
 
 [dependencies.lending-iterator-proc_macros]
 path = "src/proc_macros"
-version = "0.1.7-rc1"  # Keep in sync
+version = "0.1.7"  # Keep in sync
 
 [dev-dependencies]
 lending-iterator.path = "."

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -7,7 +7,7 @@ name = "lending-iterator"
 authors = [
     "Daniel Henry-Mantilla <daniel.henry.mantilla@gmail.com>",
 ]
-version = "0.1.6"  # Keep in sync
+version = "0.1.7-rc1"  # Keep in sync
 edition = "2021"
 
 license = "Zlib OR MIT OR Apache-2.0"
@@ -58,7 +58,7 @@ macro-vis.version = "0.1.1"
 
 [dependencies.lending-iterator-proc_macros]
 path = "src/proc_macros"
-version = "0.1.6"  # Keep in sync
+version = "0.1.7-rc1"  # Keep in sync
 
 [dev-dependencies]
 lending-iterator.path = "."

--- a/change_version.sh
+++ b/change_version.sh
@@ -2,6 +2,8 @@
 
 set -euxo pipefail
 
+git_status_clean="$(set +e; git diff-index --quiet HEAD; echo $?)"
+
 find . \
     -not -path './target/*' \
     -type f \
@@ -13,3 +15,7 @@ find . \
     \;
 
 cargo +stable update -v -w
+
+if [ "${git_status_clean}" == "0" ]; then
+    git commit -pm "Version $1 release"
+fi

--- a/change_version.sh
+++ b/change_version.sh
@@ -3,6 +3,7 @@
 set -euxo pipefail
 
 find . \
+    -not -path './target/*' \
     -type f \
     -name 'Cargo.toml' \
     -print \

--- a/src/lending_iterator/dyn/_mod.rs
+++ b/src/lending_iterator/dyn/_mod.rs
@@ -207,5 +207,5 @@ macro_rules! with_auto_traits {( $($rules:tt)* ) => (
     __emit__! { Send + Sync }
 )} pub(in crate) use with_auto_traits;
 
-#[cfg(any(doc, test))]
+#[cfg(feature = "testing")]
 mod tests;

--- a/src/lending_iterator/dyn/tests.rs
+++ b/src/lending_iterator/dyn/tests.rs
@@ -1,4 +1,3 @@
-#![allow(unused)]
 use {
     ::alloc::{
         boxed::Box,
@@ -7,11 +6,10 @@ use {
         self as lending_iterator,
         prelude::*,
     },
-    self::CanonicalHKT as Eta,
 };
 
 // Check `dyn`-unification when dealing with a non-generic way of lending.
-fn check<'r, T> (slice: &'r mut [T])
+fn _check<'r, T> (slice: &'r mut [T])
   -> Box<dyn 'r + LendingIteratorDyn<Item = HKT!(&mut [T; 1])>>
 {
     if true {
@@ -29,14 +27,14 @@ fn check<'r, T> (slice: &'r mut [T])
     }
 }
 
-fn f2<'I, I : 'I + LendingIterator + Send, Item : HKT> (i: I)
+fn _check2<'I, I : 'I + LendingIterator + Send, Item : HKT> (i: I)
 where
     I : LendingIteratorDyn<Item = CanonicalHKT<Item>>,
     // for<'any>
     //     I : LendingIteratorඞItem<'any, T = A!(Item<'any>)>
     // ,
 {
-    let mut i: Box<dyn
+    let i: Box<dyn
         'I + Send
         + LendingIteratorDyn<Item = CanonicalHKT<Item>>
     > =

--- a/src/proc_macros/Cargo.toml
+++ b/src/proc_macros/Cargo.toml
@@ -8,7 +8,7 @@ name = "lending-iterator-proc_macros"
 authors = [
     "Daniel Henry-Mantilla <daniel.henry.mantilla@gmail.com>"
 ]
-version = "0.1.6"  # Keep in sync
+version = "0.1.7-rc1"  # Keep in sync
 edition = "2021"
 
 license = "Zlib OR MIT OR Apache-2.0"

--- a/src/proc_macros/Cargo.toml
+++ b/src/proc_macros/Cargo.toml
@@ -8,7 +8,7 @@ name = "lending-iterator-proc_macros"
 authors = [
     "Daniel Henry-Mantilla <daniel.henry.mantilla@gmail.com>"
 ]
-version = "0.1.7-rc1"  # Keep in sync
+version = "0.1.7"  # Keep in sync
 edition = "2021"
 
 license = "Zlib OR MIT OR Apache-2.0"

